src/App.jsx

```diff
@@ -14,45 +18,94 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const goodsWithProperty = goodsFromServer.map((good, i) => ({
+  id: i + 1,
+  name: good,
+  length: good.length,
+}));
+
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+
+function prepeareGoods(goods, sortName, reverse) {
+  let prepearedGoods = [...goods];
+
+  if (sortName) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortName) {
+        case SORT_BY_NAME:
+          return good1.name.localeCompare(good2.name);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState('');
+  const sortedGoodsList = prepeareGoods(goodsWithProperty, sortField, reverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_BY_NAME,
+          })}
+          onClick={() => setSortField(SORT_BY_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': !reverse,
+          })}
+          onClick={() => (reverse ? setReverse('') : setReverse(REVERSE))}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-danger', 'is-light', {
+            'is-hidden': !sortField && !reverse,
+          })}
+          onClick={() => {
+            setSortField('');
+            setReverse('');
+          }}
+        >
+          Reset
+        </button>
```

Comment:
instead setting is-hidden class conditionally render or not button.
is-hidden class set display: none but button still exists in dom.
it shouldn't exists in DOM.
![Screenshot from 2023-11-08 11-49-47](https://github.com/mate-academy/react_list-of-goods-js/assets/33029864/48455d3b-8a2f-49f6-ba02-e4d54176201d)

it will fix failed test

Reply:
I think, done)